### ISSUE TITLE
Link error

### ISSUE THREAD

**Issue by pmoulon** [manhofer/Line3Dpp#2]
Thanks for having made this project has Open Source.

I just tried to compile it and I got to add imgcodecs to the list of required OpenCV libs
```
## OpenCV
FIND_PACKAGE(OpenCV REQUIRED COMPONENTS core imgproc highgui **imgcodecs**)
```

Another question, I tried on many scenes and I got very few 3D segments.
Does there is a parameter to tune in order to have more detected segments?

Here the result on the Sceaux Castle Sample scene from OpenMVG
https://github.com/openMVG/ImageDataset_SceauxCastle
![snapshot01](https://cloud.githubusercontent.com/assets/3235050/14455424/b773836a-009f-11e6-8a7f-3efdaace9c0a.png)

**Comment by manhofer** [comment]
You're welcome.

I am using OpenCV 2.4 and here the component imgcodecs apparently doesn't exist. I fixed this now in the CMakeLists.txt, so that when OpenCV 3 is used the component is included. Please try it out and tell me if it works!

Regarding the 3D lines, I suggest you try to increase the spatial regularizer (-p on the command line). Since we don't know the scale of the reconstruction (by default at least), the regularizer is defined in pixels and then projected to space (see paper for details). The default value is just 2.5px, which might sometimes be too small. Just try for instance to add "-p 5" to the command line options, which increases the allowed uncertainty to 5 pixels.

Also, when the images are rather small (as seems to be the case for this dataset) the output might not be so great, since then the line segment detection is not very accurate. I usually operate on images with more than 10 megapixels.

**Comment by pmoulon** [comment]
Thanks, I'm currently make some tests with larger images (19Mpix each) and the line count is still small.
I will make experiment with the "-p" parameter.

I will make a PR regarding an enhancement of the OpenMVG importer (in order to deal with openMVG scene that have many intrinsic).

**Comment by manhofer** [comment]
Let me know if it works. If it does not work as expected it would be great if you could share one of the datasets with me, so I can analyze it better

**Comment by pmoulon** [comment]
Thanks the fix https://github.com/manhofer/Line3Dpp/commit/77336d0ab8bb29d0caf980cc6a6129cc7dc7317a fix the error encountered with OpenCV3.

**Comment by pmoulon** [comment]
Set -p to 10 helps on this dataset, higher value reveal some outliers 
What is surprising is that it reveals all horizontal segments
![snapshot01](https://cloud.githubusercontent.com/assets/3235050/14458376/13677dc8-00b1-11e6-9f24-120c1f456591.png)

It would be super cool to compute this threshold automatically (a-contrario?)

**Comment by manhofer** [comment]
Yes, the problem with this dataset in particular is that there is only horizontal camera motion, which is bad for triangulating horizontal lines. As we can see, most of the vertical lines are already reconstructed in the earlier version, with -p 2.5 (default). This is a general issue for this kind of algorithm, and it is always recommended to have camera motion in both the vertical and the horizontal directions (if you have a choice).

I put some thought into dynamically estimating the spatial regularizer, e.g. by gradually increasing it until at least a certain percentage of the 2D lines has a valid 3D estimate. However, selecting such a percentage is non trivial and if it goes wrong you quickly end up with a lot of outliers. 

Basically, the "correct" value for -p depends on multiple factors: the image sizes, the accuracy of the camera poses, the camera motion, the precision of the line segment detector,...
It is kind of hard to make a good guess automatically, but maybe machine learning could help? I will definitely think about this in the future.

**Comment by pmoulon** [comment]
I was mentioning AContrario, since it's a way to guess threshold to find which threshold is meaningful according a distribution of residuals (In order to define how much you are close to the situation you are considering as noise).
See http://www.ipol.im/pub/art/2012/mmm-oh/
and The Likelihood-Ratio Test and Efficient Robust Estimation http://www.cv-foundation.org/openaccess/content_iccv_2015/papers/Cohen_The_Likelihood-Ratio_Test_ICCV_2015_paper.pdf

**Comment by manhofer** [comment]
Thanks, I will look into that :-)

**Comment by pmoulon** [comment]
BTW, I got nice results on another Castle dataset (Still have to increase a bit the -p threshold ;-) )
![snapshot01](https://cloud.githubusercontent.com/assets/3235050/14459491/8248b214-00b8-11e6-9511-e74030bb06a6.png)
![snapshot02](https://cloud.githubusercontent.com/assets/3235050/14459490/82486520-00b8-11e6-8605-6c14f0caff27.png)

**Comment by manhofer** [comment]
Look really nice! I'm glad that it works well
